source/agora/common/Hash.d

```diff
@@ -233,3 +203,65 @@ nothrow @nogc @safe unittest
     str.c2 = 'c';
     assert(hashFull(str) == abc_exp);
 }
+
+/*******************************************************************************
+
+    Hashes multiple arguments together
+
+    Params:
+        T = variadic argument types
+        args = the arguments
+
+    Returns:
+        the hash of all the arguments
+
+*******************************************************************************/
+
+public Hash hashMulti (T...)(T args) nothrow @nogc @safe
+{
+    static struct HashMulti
+    {
+        T fields;
+
+        public void computeHash (scope HashDg dg) const nothrow @safe @nogc
+        {
+            foreach (const ref field; this.fields)
```

Comment:
Whouldn't we be able to do:
```d
static foreach (idx, ref _; args)
    hashPart(args[idx], dg);
```

Or something similar ?

Reply:
Yep.

Reply:
Ok so I can do it, but only if I make the struct non-static. And that means:

```
source/agora/common/Hash.d(220,13): Error: function `agora.common.Hash.hashMulti!(BitBlob!512LU, BitBlob!512LU).hashMulti` is @nogc yet allocates closures with the GC
```

So I think s/Yep/Nope.

Reply:
I mean you can foreach on the tuple, so you don't need the struct to begin with.

Reply:
I could duplicate some of the stuff done in hashFull like:

```D
    Hash hash = void;
    crypto_generichash_state state;
    crypto_generichash_init(&state, null, 0, Hash.sizeof);
    scope HashDg dg = (scope const(ubyte)[] data) @trusted
                     => cast(void)crypto_generichash_update(&state, data.ptr, data.length);
    hashPart(record, dg);
    crypto_generichash_final(&state, hash[].ptr, Hash.sizeof);
    return hash;
```

That could work.